# Problem Statement 4-->

Telecall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and must be reworked before processing. The manager wants to check whether the defective % varies by center. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

File: Customer OrderForm.csv

# Business Objectives-->


# Constraints-->


In [2]:
#load a required libraries
import pandas as pd
import numpy as np
import scipy 
from scipy import stats

In [3]:
#Load a dataset
df=pd.read_csv("CustomerOrderForm.csv")

In [4]:
#print 1st 5 rows of dataset
df.head(5)

,Phillippines,Indonesia,Malta,India
0,Error Free,Error Free,Defective,Error Free
1,Error Free,Error Free,Error Free,Defective
2,Error Free,Defective,Defective,Error Free
3,Error Free,Error Free,Error Free,Error Free
4,Error Free,Error Free,Defective,Error Free


In [5]:
#print information of dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 315 entries, 0 to 314
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 10.0+ KB


In [6]:
#print shape of dataset
df.shape

(315, 4)

In [7]:
# Convert categorical values into binary (Error Free = 0, Defective = 1)
df.replace({"Error Free": 0, "Defective": 1}, inplace=True)

In [8]:
df.columns

Index(['Phillippines', 'Indonesia', 'Malta', 'India'], dtype='object')

In [9]:
df.head(1)

,Phillippines,Indonesia,Malta,India
0,0.0,0.0,1.0,0.0


In [10]:
# Create a contingency table (counts of defective and error-free)
contingency_table = df.apply(pd.Series.value_counts).T

In [11]:
print("\nContingency Table (Defective vs. Error-Free):")
print(contingency_table)


Contingency Table (Defective vs. Error-Free):
              0.0  1.0
Phillippines  271   29
Indonesia     267   33
Malta         269   31
India         280   20


In [12]:
# Perform Chi-Square Test for Independence
chi2_stat, p_value, dof, expected = stats.chi2_contingency(contingency_table)

In [13]:
# Display results
print("\nChi-Square Test Results:")
print(f"Chi-Square Statistic = {chi2_stat:.4f}")
print(f"p-value = {p_value:.4f}")
print(f"Degrees of Freedom = {dof}")
print("\nExpected Frequencies (if no significant difference):")
print(expected)


Chi-Square Test Results:
Chi-Square Statistic = 3.8590
p-value = 0.2771
Degrees of Freedom = 3

Expected Frequencies (if no significant difference):
[[271.75  28.25]
 [271.75  28.25]
 [271.75  28.25]
 [271.75  28.25]]


In [14]:
# Interpretation based on significance level alpha = 0.05
alpha = 0.05  
if p_value < alpha:
    print("\nConclusion: Reject H0. The defective % varies significantly by center.")
else:
    print("\nConclusion: Fail to reject H0. No significant difference in defective % among centers.")


Conclusion: Fail to reject H0. No significant difference in defective % among centers.


Interpretation of Chi-Square Test Results:

Result: p-value = 0.2771

Interpretability:
H0 (Null Hypothesis): The proportion of defective order forms is the same across all four centers.

H1 (Alternative Hypothesis): The proportion of defective order forms differs among the four centers.

Since the p-value (0.2771) > 0.05, we fail to reject the null hypothesis.

Conclusion:
There is no significant difference in defect rates among the four centers.
The defects appear to be randomly distributed rather than associated with a specific center.